src/Dotnet.Script.Core/ScriptCompiler.cs

```diff
@@ -190,8 +190,12 @@ private Assembly MapUnresolvedAssemblyToRuntimeLibrary(IDictionary<string, Runti
             {
                 if (runtimeAssembly.Name.Version > assemblyName.Version)
                 {
-                    Logger.Log($"Redirecting {assemblyName} to {runtimeAssembly.Name}");                    
-                    return Assembly.LoadFile(runtimeAssembly.Path);
+                    var loadedAssembly = AppDomain.CurrentDomain.GetAssemblies()
```

Comment:
we will probably need to find a more efficient way to do it; it should be possible to attach to the `Resolving` event of `AssemblyLoadContext` and keep track of loaded assemblies rather than enumerate all the time

Reply:
If I remember correctly, the `MapUnresolvedAssemblyToRuntimeLibrary` does not get executed very often. Currently we have only one test that fails if we omit the handler all together. Not being a "hot path" I'll say that we can live with the enumeration for now 👍

Reply:
@filipw do you want me to do something on this one?